### PR TITLE
`Adaptive learning`: Add standardized competency management

### PR DESCRIPTION
**Pull request by rstief**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I added pre-authorization annotations according to the [guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [x] I documented the Java code using JavaDoc style.

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
Adds the functional requirement 2.2 described in #8001

### Description
Adds the possibility to view all standardized competencies in a tree structure. Allows to filter by competency title and knowledge area.
In this view, standardized competencies can be created, edited and deleted.

### Steps for Testing
Prerequisites:
- **ONLY TEST ON TS2!!! OTHER TEST SERVERS MAY NOT HAVE KNOWLEDGE AREAS!**
- 1 Admin Account

1. Go to Server Administration -> Feature Toggles and activate Standardized Competencies
2. Go to Server Administration -> Standardized Competencies
3. Test that the filtering by title works, only competencies containing the filter string should show
4. Test that the filtering by knowledge area works, only competencies in that area **and direct ancestors** should show
5. Test inserting/deleting competencies, see that the tree updates
6. Test updating a competency:
  1. general updates
  2. test that updating the title to not match the filter anymore makes the competency "dissapear" (it should still re-appear after removing the filter) 
  3. test that moving a competency to a knowledge area not in the filter also makes it "dissapear"

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [admin-standardized-competency.service.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6281/latest/artifact/shared/Coverage-Report-Client-Tests/app/admin/standardized-competencies/admin-standardized-competency.service.ts.html) | 100% | ✅ |
| [standardized-competency-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6281/latest/artifact/shared/Coverage-Report-Client-Tests/app/admin/standardized-competencies/standardized-competency-detail.component.ts.html) | 100% | ✅ |
| [standardized-competency-management.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6281/latest/artifact/shared/Coverage-Report-Client-Tests/app/admin/standardized-competencies/standardized-competency-management.component.ts.html) | 91.39% | ✅  |
| [standardized-competency.service.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6281/latest/artifact/shared/Coverage-Report-Client-Tests/app/admin/standardized-competencies/standardized-competency.service.ts.html) | 100% | ✅ |
| [standardized-competency.model.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6281/latest/artifact/shared/Coverage-Report-Client-Tests/app/entities/competency/standardized-competency.model.ts.html) | 93.75% | ✅ |
| [feature-toggle.service.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6281/latest/artifact/shared/Coverage-Report-Client-Tests/app/shared/feature-toggle/feature-toggle.service.ts.html) | 67.56% | ✅  (unchanged) |
| [navbar.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6281/latest/artifact/shared/Coverage-Report-Client-Tests/app/shared/layouts/navbar/navbar.component.ts.html) | 88.85% | ✅ (unchanged) |

#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [BaseCompetency.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6281/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain.competency/BaseCompetency.html) | 100% | ✅ |
| [Competency.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6281/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain.competency/Competency.html) | 79% | ✅  |
| [KnowledgeArea.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6281/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain.competency/KnowledgeArea.html) | 100% | ✅ |
| [Source.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6281/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain.competency/Source.html) | 92% | ✅  |
| [StandardizedCompetency.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6281/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain.competency/StandardizedCompetency.html) | 89% | ✅  |
| [SourceRepository.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6281/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.repository/SourceRepository.html) | 100% | ✅ |
| [KnowledgeAreaRepository.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6281/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.repository.competency/KnowledgeAreaRepository.html) | 100% | ✅ |
| [StandardizedCompetencyRepository.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6281/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.repository.competency/StandardizedCompetencyRepository.html) | 100% | ✅ |
| [KnowledgeAreaService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6281/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.competency/KnowledgeAreaService.html) | 100% | ✅ |
| [StandardizedCompetencyService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6281/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.competency/StandardizedCompetencyService.html) | 91% | ✅  |
| [Feature.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6281/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.feature/Feature.html) | 100% | ✅ |
| [FeatureToggleService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6281/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.feature/FeatureToggleService.html) | 100% | ✅ |
| [AdminStandardizedCompetencyResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6281/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest.admin/AdminStandardizedCompetencyResource.html) | 100% | ✅ |
| [StandardizedCompetencyResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6281/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest.competency/StandardizedCompetencyResource.html) | 100% | ✅ |
| [KnowledgeAreaDTO.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6281/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest.dto.competency/KnowledgeAreaDTO.html) | 100% | ✅ |
| [StandardizedCompetencyDTO.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6281/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest.dto.competency/StandardizedCompetencyDTO.html) | 100% | ✅ |

### Screenshots
Detail View:
image
Edit View:
image

## Summary by CodeRabbit

- **New Features**
	- Introduced standardized competencies and knowledge areas management in the admin module.
	- Added CRUD operations for standardized competencies and knowledge areas.
	- Implemented hierarchical structure retrieval for knowledge areas.
	- Enabled feature toggle for standardized competencies.
- **Bug Fixes**
	- Adjusted `KnowledgeArea` data model to correctly handle parent-child relationships.
- **Documentation**
	- Added documentation for new DTOs related to standardized competencies and knowledge areas.
- **Refactor**
	- Updated service and repository layers to support new competency features.
- **Tests**
	- Added comprehensive integration and unit tests for standardized competency features.
- **Style**
	- Introduced new styling for the standardized competency management component.
- **Chores**
	- Enhanced feature toggle service to include standardized competencies.

### CODE REVIEW

**Comment by rstief** [comment]
see if this helps deploy :)

**Review by laurenzfb — APPROVED** [review]
tested on ts2. Everything works as described

File: src/main/webapp/app/admin/standardized-competencies/standardized-competency-detail.component.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    }
+
+    @Input() dialogError: Observable<string>;
+
+    @Output() onSave = new EventEmitter<StandardizedCompetencyDTO>();
+    @Output() onDelete = new EventEmitter<void>();
+    @Output() onClose = new EventEmitter<void>();
+    @Output() isInEditModeChange = new EventEmitter<boolean>();
+
+    private _isInEditMode: boolean;
```
Comment (nityanandaz) [inline src/main/webapp/app/admin/standardized-competencies/standardized-competency-detail.component.ts:54]: Could be `isEditing`, compare https://developer.apple.com/documentation/uikit/uiviewcontroller/1621498-isediting

File: src/main/webapp/i18n/de/standardizedCompetency.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{
+    "artemisApp": {
+        "standardizedCompetency": {
+            "title": "Standardisierte Kompetenzen",
+            "manage": {
+                "title": "Standardisierter Kompetenzen Verwalten",
```
Comment (nityanandaz) [inline src/main/webapp/i18n/de/standardizedCompetency.json:6]:
```suggestion
                "title": "Standardisierte Kompetenzen Verwalten",
```

**Review by nityanandaz — DISMISSED** [review]
Code review: minor comments :)

**Review by JohannesWt — CHANGES_REQUESTED** [review]

**Review by JohannesWt — APPROVED** [review]
Code looks good to me after changes

**Review by mateusmm01 — DISMISSED** [review]
Demonstrated in testing session and works as expected!

**Review by laurenzfb — APPROVED** [review]
reapprove after small improvements

**Review by nityanandaz — DISMISSED** [review]
Code review: re-approve

**Review by MaximilianAnzinger — CHANGES_REQUESTED** [review]
Please check the GH Actions (e.g. Atlassian Tests & Server Style are failing)

**Review by MaximilianAnzinger — APPROVED** [review]
Thx for fixing the remaining style issues 👍

**Pull request merged**